### DISCUSSION TITLE
Solved: Variable name doesn't work with Sync?

### DISCUSSION THREAD

**Discussion by Taitava** [Taitava/obsidian-shellcommands#137] [Debugging and testing]
I created this discussion based on issue #136. Quotes from the issue:

@Mara-Li https://github.com/Taitava/obsidian-shellcommands/issues/136#issue-1115340408
> Variable doesn't work :
> 
>   * Variable prevent to open the command in modal
>   * Variable seems to break the last character when adding the last `}}`
>     ![image](https://user-images.githubusercontent.com/30244939/151222239-92ed76c2-909c-4d64-a053-00f87d657383.png)
>     ![image](https://user-images.githubusercontent.com/30244939/151222345-96d6c595-3baa-4191-a031-991c80a3c6b9.png)
>     ![image](https://user-images.githubusercontent.com/30244939/151222502-d023d0e5-9c89-4230-9f40-d71f84a12b7f.png)

@Mara-Li https://github.com/Taitava/obsidian-shellcommands/issues/136#issuecomment-1022588455
> I try a lot around the plugin, and it seems to break sync.
> 
> It worked well before, but now I can't really make it work. It works without problem on a test vault.

**Comment by Taitava** [comment]
Thanks for your report! And thanks for the helpful screenshots!

Judging from the fact that the shell command preview texts show incomplete variable names, I guess someting crashes when the variables are being parsed for the review text, and also when opening the command palette. By crashing I mean that there probably happens some kind of unhandled exception/error, and when it happens, it stops part of JavaScript execution, leading to things like incomplete preview texts, shell commands not appearing in the command palette, and maybe other issues.

I'd ask you for the following things:
1. Can you provide the _Shell commands_ plugin's version number, please?
2. Can you open up _Obsidian Developer Console_ (`Ctrl`/`Cmd` + `Shift` + `I`) right after this issue has happened, and then try to see if you can see some error messages in the console? If there are error messages, can you take screenshots of them, please?
3. When you say _Sync_, do you mean the [Obsidian Sync](https://obsidian.md/sync) service, or something else?
4. Can you also try to disable all other community plugins, so that the only plugins you have enabled are: _Shell commands_ plugin, Sync, and any possible core plugins you use. So, no need to disable any core plugins, only community plugins. Then try to see if the problem still persists.

**Reply by Mara-Li** [reply]
Hello!
I will do all of this today, on another sync vault. The first time things happened, all my files became corrupted. By chance, I have a lot of save but now I'm afraid.

**Comment by Mara-Li** [comment]
1. Last version of Shell Commands : 0.9.0
2. Yes, there is an error !
![image](https://user-images.githubusercontent.com/30244939/151388797-172fde8a-fa56-4fff-bb60-b7ae234f7684.png)

2. Removing sync + reload : 
![image](https://user-images.githubusercontent.com/30244939/151389489-2b8dd604-b792-42f4-bbfc-cf7088834af8.png)

→ Adding Sync + active : 
![image](https://user-images.githubusercontent.com/30244939/151389666-66964045-3025-4f52-b92e-2282d7e2e368.png)

So, sync doesn't seems to change anything.

Activation of all plugin I use : 
![image](https://user-images.githubusercontent.com/30244939/151390085-8f9180fd-56ba-4226-81e6-9d1e98d31874.png)
![image](https://user-images.githubusercontent.com/30244939/151390145-21d14a12-de76-431a-9a8c-5018dad704a9.png)

Seems a plugin can cause this bug.


So, after TRYING 
ONE
BY
ONE
My 70 plugin, I found the plugin that make a crash of your.
It was "Dynamic Highlight", of nothingislost : https://github.com/nothingislost/obsidian-dynamic-highlights

**Reply by Mara-Li** [reply]
Also, someone spotted the issue before me on Dynamic Hightlights : https://github.com/nothingislost/obsidian-dynamic-highlights/issues/25

**Reply by Taitava** [reply]
Thank you for giving more information and for the link to the issue in Dynamic Highlights. I didn't yet have time to look into this problem, but I'll keep an eye on the issue and will see if there's something for me to fix in my plugin. :+1:

**Comment by Taitava** [comment]
# Probably solved

[Updating _Dynamic highlights_ to version `0.2.2` solves this issue](https://github.com/nothingislost/obsidian-dynamic-highlights/issues/25#issuecomment-1039210919). However, there's also something to be fixed in SC's side too (to prevent similar issues happening in the future), for which I created a new issue #152.

Please let me know if you still encounter problems after updating DH. Remember to restart Obsidian after the update! 🙂
